pkg/v1/remote/check.go

```diff
@@ -24,7 +24,7 @@ func CheckPushPermission(ref name.Reference, kc authn.Keychain, t http.RoundTrip
 	}
 
 	scopes := []string{ref.Scope(transport.PushScope)}
-	tr, err := transport.New(ref.Context().Registry, auth, t, scopes)
+	tr, err := transport.NewWithContext(context.Background(), ref.Context().Registry, auth, t, scopes)
```

Comment:
Should be `context.TODO` probably? (We want to remove this func eventually given that TODO, but in case we don't, it's better to track this lack of ctx propagation)

Reply:
You could probably say the same about the `context.Background` on line 40 😄 . Should I change that as well or leave it alone?